app/Models/Behaviors/HasApiRelations.php

```diff
@@ -106,8 +106,11 @@ public function getLocalApiMapping($items, $apiElements)
     {
         // Find locally selected objects
         return $items->filter(function ($relatedElement) use ($apiElements) {
-            $apiRelationElement = \App\Models\ApiRelation::where('id', $relatedElement->related_id)->first();
-            $result = $apiElements->where('id', $apiRelationElement->datahub_id)->first();
+            $apiRelationElement = \App\Models\ApiRelation::where('id', $relatedElement->related_id)
+                ->first()
+                ?->datahub_id;
+
+            $result = $apiElements->where('id', $apiRelationElement)->first();
```

Comment:
Very minor, and I know you didn't write this originally, but `Model::where('id', $id)->first()` is a long way of writing `Model::find($id)`. This could be shortnened to:
```php
$datahubId = \App\Models\ApiRelation::find($relatedElement->related_id)?->datahub_id;
$result = $apiElements->find($datahubId);
```

Reply:
It looks like the ApiCollection model doesn't have a find method which is really odd? 
<img width="810" alt="Screenshot 2024-11-13 at 10 57 40 AM" src="https://github.com/user-attachments/assets/c02fa808-e107-449e-ad28-8e7f28384b8d">

Reply:
Weird, oh well 🤷